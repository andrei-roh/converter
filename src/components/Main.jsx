import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { fetchBelarusRubleRate } from './redux/actions';
import Spinner from './Spinner/Spinner';
import Error from './Error/Error';
import Content from './Content/Content';
import { ThemeProvider } from 'styled-components';
import { GlobalStyles } from './globalStyles';
import { lightTheme, darkTheme } from './themes';

const Main = ({
  loading,
  error,
  defaultValue,
  belarusRuble,
  belarusRubleToOther,
  onfetchBelarusRubleRate,
}) => {
  const [theme, setTheme] = useState('light');
  const [showDark, setShowDark] = useState(false);
  const handleShowDark = (showDark) => {
    setShowDark(!showDark);
    theme === 'light' ? setTheme('dark') : setTheme('light');
  };
  useEffect(() => {
    onfetchBelarusRubleRate();
  }, [onfetchBelarusRubleRate]);

  return (
    <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme}>
      <GlobalStyles />
      {loading ? (
        <Spinner />
      ) : error ? (
        <Error error={error} />
      ) : (
        <Content
          defaultValue={defaultValue}
          belarusRuble={belarusRuble}
          belarusRubleToOther={belarusRubleToOther}
          showDark={showDark}
          handleShowDark={handleShowDark}
        />
      )}
    </ThemeProvider>
  );
};

const mapStateToProps = (state) => ({
  loading: state.loading,
  error: state.error,
  defaultValue: state.defaultValue,
  belarusRuble: state.belarusRuble,
  belarusianRubleRate: state.belarusianRubleRate,
  belarusRubleToOther: state.belarusRubleToOther,
});

const mapDispatchToProps = (dispatch) => ({
  onfetchBelarusRubleRate: () => dispatch(fetchBelarusRubleRate()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Main);
