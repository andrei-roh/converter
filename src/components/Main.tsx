import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { fetchBelarusRubleRate } from './redux/actions';
import Spinner from './Spinner/Spinner';
import Error from './Error/Error';
import Content from './Content/Content';
import { ThemeProvider } from 'styled-components';
import { GlobalStyles } from './globalStyles';
import { lightTheme, darkTheme } from './themes';
import { IMain, State } from '../types';

const Main: React.FC<IMain> = ({
  loading,
  error,
  defaultValue,
  belarusRuble,
  belarusRubleToOther,
  onfetchBelarusRubleRate,
  theme,
  endpoint,
}) => {
  useEffect(() => {
    onfetchBelarusRubleRate(endpoint);
  }, [endpoint, onfetchBelarusRubleRate]);

  return (
    <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme}>
      <GlobalStyles />
      {loading ? (
        <Spinner />
      ) : error ? (
        <Error error={error} />
      ) : (
        <Content
          defaultValue={defaultValue}
          belarusRuble={belarusRuble}
          belarusRubleToOther={belarusRubleToOther}
        />
      )}
    </ThemeProvider>
  );
};

const mapStateToProps = (state: State) => ({
  loading: state.loading,
  error: state.error,
  defaultValue: state.defaultValue,
  belarusRuble: state.belarusRuble,
  belarusRubleToOther: state.belarusRubleToOther,
  theme: state.theme,
  showDark: state.showDark,
  endpoint: state.endpoint,
});

const mapDispatchToProps = (dispatch: any) => ({
  onfetchBelarusRubleRate: (url: string) =>
    dispatch(fetchBelarusRubleRate(url)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Main);
