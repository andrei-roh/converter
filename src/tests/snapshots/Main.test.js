import { shallow } from 'enzyme';
import Main from '../../components/Main';
import store from '../../components/redux/store';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Main store={store} />);
});

describe('<Main /> rendering', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
