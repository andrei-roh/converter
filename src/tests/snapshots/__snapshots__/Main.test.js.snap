// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> rendering renders correctly 1`] = `ShallowWrapper {}`;
